pkg/config/nodetreemodel/config.go

```diff
@@ -435,32 +435,52 @@ func (c *ntmConfig) mergeAllLayers() error {
 
 	c.root = root
 	// recompile allSettings now that we have the full config
-	c.allSettings = c.computeAllSettings(c.schema, "")
+	c.allSettings = c.computeAllSettings("")
 	return nil
 }
 
-func (c *ntmConfig) computeAllSettings(node InnerNode, path string) []string {
+func (c *ntmConfig) computeAllSettings(path string) []string {
 	c.maybeRebuild()
 
-	knownKeys := []string{}
+	keySet := make(map[string]struct{})
+
+	// 1. Collect all known keys from schema
+	c.collectKeysFromNode(c.schema, path, keySet, true)
+
+	// 2. Collect all keys from merged tree (only ones with values)
+	c.collectKeysFromNode(c.root, "", keySet, false)
+
+	// 3. Collect all unknown keys, even if they have no value set
+	c.collectKeysFromNode(c.unknown, "", keySet, true)
+
+	allKeys := make([]string, 0, len(keySet))
+	for key := range keySet {
+		allKeys = append(allKeys, key)
+	}
+
+	slices.Sort(allKeys)
+	return allKeys
+}
+
+func (c *ntmConfig) collectKeysFromNode(node InnerNode, path string, keySet map[string]struct{}, includeAllSchemaKeys bool) {
 	for _, name := range node.ChildrenKeys() {
 		newPath := joinKey(path, name)
 
-		child, _ := node.GetChild(name)
+		child, err := node.GetChild(name)
+		if err != nil {
+			continue
+		}
+
 		if leaf, ok := child.(LeafNode); ok {
-			if leaf.Source() != model.SourceSchema {
-				knownKeys = append(knownKeys, newPath)
-			} else if c.leafAtPathFromNode(newPath, c.root) != missingLeaf {
-				knownKeys = append(knownKeys, newPath)
+			// For schema nodes, include all keys if requested
```

Comment:
```suggestion
			// Include all keys if requested
```

(as noted above, we also end up here for the unknown set, which is not the schema)